### PR TITLE
[Job Launcher] Added job details endpoint

### PR DESCRIPTION
**Pull request by eugenvoronov**
## Description
Created a jod details endpoint for Fortune type of job.

**Implementation details** 
Endpoint should return the information for this UX https://www.figma.com/file/u81GKebVavmQW8zGHb5Jgo/Dashboards---Everything-Apps?type=design&node-id=1894-16431&mode=design&t=ZJ4dHTfPZdOxavFo-0

## Summary of changes

- [x] Add job details endpoint 
- [x] Add logic for Fortune job type
- [x] Create unit tests
- [x] Add logic for CVAT job type
- [ ] Retrieve slash tokens  
- [x] Implement paid out mechanism

## How test the changes
`GET /job/details/:jobId`
`yarn test`

## Related issues
[[Job Launcher] Escrow details endpoint #863](https://github.com/humanprotocol/human-protocol/issues/863)

### CODE REVIEW

File: packages/apps/job-launcher/server/src/modules/job/job.service.ts [hunk 1]
```diff
@@ -643,4 +647,124 @@ export class JobService {
...
+        requesterAddress: signer.address,
+        exchangeOracleAddress,
+        recordingOracleAddress,
+        reputationOracleAddress
+      }
+    } else {
+      manifestDetails = {
+        chainId,
+        requestType: (manifest as CvatManifestDto).annotation.type,
+        submissionsRequired: (manifest as CvatManifestDto).annotation.job_size,
```
Comment (Dzeranov) [inline packages/apps/job-launcher/server/src/modules/job/job.service.ts:710]: @eugenvoronov job size reflects an amount of jobs the entire dataset will be split. To reflect the number of sumbissions you need to retrieve a dataset size.

File: packages/apps/job-launcher/server/src/modules/job/job.service.ts [hunk 2]
```diff
@@ -643,4 +647,124 @@ export class JobService {
...
+        tokenAddress,
+        fundAmount: Number((manifest as CvatManifestDto).job_bounty),
```
Comment (Dzeranov) [inline packages/apps/job-launcher/server/src/modules/job/job.service.ts:712]: `job_bounty` reflects an approx reward user will receive for annotating a job (part of the dataset)

File: packages/apps/job-launcher/server/src/common/interfaces/job.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { JobStatus } from '../enums/job';
+import { JobRequestType, JobStatus } from '../enums/job';
```
Comment (portuu3) [inline packages/apps/job-launcher/server/src/common/interfaces/job.ts:1]: unnecessary import

File: packages/apps/job-launcher/server/src/common/utils/index.ts [hunk 4]
```diff
@@ -70,3 +70,27 @@ export const parseUrl = (
 
   throw new Error('Invalid URL');
 };
+
+export function parseMinioURL(url: string) {
```
[the remark this answers is not in the document]
Comment (portuu3) [reply packages/apps/job-launcher/server/src/common/utils/index.ts:74]: Agree with @flopez7 if the mentioned function can be reused

File: packages/apps/job-launcher/server/src/modules/job/job.service.ts [hunk 5]
```diff
@@ -643,4 +647,134 @@ export class JobService {
 
     return true;
   }
+
+  public async getDetails(userId: number, jobId: number): Promise<JobDetailsDto> {
```
Comment (portuu3) [inline packages/apps/job-launcher/server/src/modules/job/job.service.ts:651]: I think we can use subgraph to get all the data related to escrow, oracles, balance, token, etc .
We will need to add function to the sdk that returns the same as [this](https://sdk.humanprotocol.org/human-protocol-sdk/js-ts/escrow/escrowclient/getescrows) but called getEscrow and we pass the address as parameter

**Review by portuu3 — CHANGES_REQUESTED** [review]

File: packages/apps/job-launcher/server/src/modules/job/job.service.ts [hunk 6]
```diff
@@ -643,4 +647,134 @@ export class JobService {
...
+        submissionsRequired: (manifest as FortuneManifestDto).submissionsRequired,
+        tokenAddress,
+        fundAmount: jobEntity.fundAmount,
+        requesterAddress: signer.address,
+        exchangeOracleAddress,
+        recordingOracleAddress,
+        reputationOracleAddress
+      }
+    } else if (CVAT_JOB_TYPES.includes((manifest as CvatManifestDto).annotation.type)) {
+      const { useSSL, host, port, bucket } = parseMinioURL((manifest as CvatManifestDto).data.data_url)
```
[the remark this answers is not in the document]
Comment (eugenvoronov) [reply packages/apps/job-launcher/server/src/modules/job/job.service.ts:707]: For our case, if we have:
```
bucket/dataset1/
bucket/dataset2/
```
... and we just use listObjects(bucket), we would end up counting objects from both `dataset1` and `dataset2`.

To handle such scenarios, we would need to specify a prefix when listing objects. 
```
const dataset = await storageClient.listObjects(bucket, 'dataset1/');
```

But for this we need to update our SDK:
```
this.client.listObjectsV2(bucket, <DIR_NAME>, true, '');
```

File: packages/apps/job-launcher/server/src/modules/job/job.service.ts (hunk 5 above)
Comment (eugenvoronov) [reply packages/apps/job-launcher/server/src/modules/job/job.service.ts:651]: Moved in the separate issue [[Job Launcher] [SDK] Implement getEscrow method to SDK and use it in getDetails method#892](https://github.com/humanprotocol/human-protocol/issues/892)

File: packages/apps/job-launcher/server/src/modules/job/job.service.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (eugenvoronov) [reply packages/apps/job-launcher/server/src/modules/job/job.service.ts]: Moved in the separate issue [[Job Launcher] [SDK] Add directory param to listObjects method in SDK#893](https://github.com/humanprotocol/human-protocol/issues/893)

**Review by portuu3 — APPROVED** [review]

**Pull request merged**